nibiru/query_clients/util.py

```diff
@@ -16,16 +30,75 @@ def t_dict(d):
     }
 
 
-def deserialize(proto_message: object) -> dict:
+def deserialize(pb_msg: protobuf_message.Message) -> dict:
+    """Deserializes a proto message into a dictionary.
+
+    - sdk.Dec values are converted to floats.
+    - sdk.Int values are converted to ints.
+    - Missing fields become blank strings.
+
+    Args:
+        pb_msg (protobuf.message.Message)
+
+    Returns:
+        dict: 'pb_msg' as a JSON-able dictionary.
+    """
+    if not isinstance(pb_msg, protobuf_message.Message):
+        raise TypeError(f"expted protobuf Message for 'pb_msg', not {type(pb_msg)}")
+    custom_dtypes: Dict[str, bytes] = {
+        str(field[1]): field[0].GetOptions().__getstate__().get("serialized", None)
+        for field in pb_msg.ListFields()
+    }
+    serialized_output = {}
+    expected_fields: List[str] = [
+        attr for attr in dir(pb_msg) if attr not in PROTOBUF_MSG_BASE_ATTRS
+    ]
+
+    for _, attr in enumerate(expected_fields):
+
+        attr_search = pb_msg.__getattribute__(attr)
+        custom_dtype = custom_dtypes.get(str(attr_search))
+
+        if custom_dtype is not None:
+
+            if "sdk/types.Dec" in str(custom_dtype):
+                serialized_output[str(attr)] = from_sdk_dec(
+                    pb_msg.__getattribute__(attr)
+                )
+            elif "sdk/types.Int" in str(custom_dtype):
+                serialized_output[str(attr)] = from_sdk_int(
+                    pb_msg.__getattribute__(attr)
+                )
+            else:
+                try:
+                    val = pb_msg.__getattribute__(attr)
+                    if hasattr(val, '__len__') and not isinstance(val, str):
+                        updated_vals = []
+                        for v in val:
+                            updated_vals.append(deserialize(v))
+                        serialized_output[str(attr)] = updated_vals
+                    else:
+                        serialized_output[str(attr)] = deserialize(val)
+                except:
+                    serialized_output[str(attr)] = pb_msg.__getattribute__(attr)
+        elif (custom_dtype is None) and (attr_search == ''):
+            serialized_output[str(attr)] = ""
+        else:
+            serialized_output[str(attr)] = deserialize(pb_msg.__getattribute__(attr))
+
+    return serialized_output
+
+
+def deserialize_exp(proto_message: protobuf_message.Message) -> dict:
```

Comment:
Do we still need this function?

Reply:
I figured that you may have added functionality here and wasn't sure if I should remove it.

Reply:
The logic of `deserialize_exp` is cleaner, so I'd like to try to make the current `deserialize` more simple in a refactor.